src/rgw/CMakeLists.txt

```diff
@@ -348,6 +348,8 @@ target_link_libraries(rgw_schedulers
 
 set(radosgw_srcs
   rgw_main.cc
+  rgw_file.cc
+  librgw.cc
```

Comment:
why are these compiled into the `radosgw` binary now?

Reply:
circular dependencies--I'll revisit and see if there's an alternative

Reply:
ah. would it help to move AppMain out of rgw_main.cc?

i think we should avoid linking a main() function into librgw

Reply:
most likely;  it just seems inapropos, but most likely rgw_appmain.cc would do it